### PR TITLE
various housecleaning tidbits

### PR DESCRIPTION
**Pull request by MWGNZ**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,5 @@ dist
 client/statik/
 client/version.go
 Thumbs.db
+*.lock
```
Comment (Fugiman) [inline .gitignore]: aren't yarn.lock files important?

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,5 @@ dist
...
+go/
```
Comment (Fugiman) [inline .gitignore]: What does this apply to?

File: build.sh [hunk 3]
```diff
@@ -10,14 +11,15 @@
...
+CWD=$(pwd)
 rm -rf dist
 mkdir dist
 mkdir dist/twitch
 mkdir dist/netlify
 
 # Build twitch bundle
-cd config && yarn build && cd ..
-cd overlay && yarn build && cd ..
+cd config && npm install && yarn build && cd ..
```
Comment (Fugiman) [inline build.sh]: I'd prefer `yarn install` over `npm install`

File: build.sh [hunk 4]
```diff
@@ -28,15 +30,16 @@ cd dist/twitch && zip -r ../twitch.zip * && cd ../..
...
 echo $VERSION > dist/netlify/version
 
-go get github.com/karalabe/xgo
 export DOCKER_HOST=192.168.99.100:2376
-export GOPATH=/c/Users/fugi_000/go
-xgo --targets=windows/amd64 -ldflags="-H windowsgui" -out deckmaster -dest /c/Users/fugi_000/go/src/github.com/fugiman/deckmaster/dist/netlify ./client
+export GOPATH=${HOME}/go
+go get github.com/karalabe/xgo
+
+$GOPATH/bin/xgo --targets=windows/amd64 -ldflags="-H windowsgui" -out deckmaster -dest $CWD/dist/netlify ./client
```
Comment (Fugiman) [inline build.sh]: So amusingly I use Linux Subsystem for Windows for my development, which means that `${HOME}` is not `/c/Users/fugi_000/go`. and in fact `/c` is a symlink to `/mnt/c` because Docker is running in windows and not linux because I have Win 10 Home, not Pro, so I can't install docker native and really everything's a mess.

tldr this change will horribly break my personal dev environment. I have no idea what it'll do to yours. I'd be interested in discussing how to improve this process (vagrant?) but I don't think this is the solution.

File: .gitignore (hunk 1 above)
Comment (MWGNZ) [reply .gitignore]: honestly I don't know, happy to revert this bit

File: .gitignore (hunk 2 above)
Comment (MWGNZ) [reply .gitignore]: adding in the go folder means we ignore what is now the default GOPATH

File: build.sh (hunk 3 above)
Comment (MWGNZ) [reply build.sh]: I have no idea what the difference is, only that adding in npm install allowed me to progress further through the build process. more than happy to change it

File: build.sh (hunk 4 above)
Comment (MWGNZ) [reply build.sh]: that's cool. the main thing I was going for here was some sort of generic that doesn't use your username, but it may be easier to just leave as is.
